File: csharp-example/csharp-example/app/Application.cs
using System;
using System.Collections.Generic;
using OpenQA.Selenium;
using OpenQA.Selenium.Chrome;

namespace csharp_example
{
    public class Application
    {
        private IWebDriver driver;

        private RegistrationPage registrationPage;
        private AdminPanelLoginPage adminPanelLoginPage;
        private CustomerListPage customerListPage;

        public Application()
        {
            driver = new ChromeDriver();
            registrationPage = new RegistrationPage(driver);
            adminPanelLoginPage = new AdminPanelLoginPage(driver);
            customerListPage = new CustomerListPage(driver);
        }

        public void Quit()
        {
            driver.Quit();
        }

        internal void RegisterNewCustomer(Customer customer)
        {
            registrationPage.Open();
            registrationPage.FirstnameInput.SendKeys(customer.Firstname);
            registrationPage.LastnameInput.SendKeys(customer.Lastname);
            registrationPage.Address1Input.SendKeys(customer.Address);
            registrationPage.PostcodeInput.SendKeys(customer.Postcode);
            registrationPage.CityInput.SendKeys(customer.City);
            registrationPage.SelectCountry(customer.Country);
            registrationPage.SelectZone(customer.Zone);
            registrationPage.EmailInput.SendKeys(customer.Email);
            registrationPage.PhoneInput.SendKeys(customer.Phone);
            registrationPage.PasswordInput.SendKeys(customer.Password);
            registrationPage.ConfirmedPasswordInput.SendKeys(customer.Password);
            registrationPage.CreateAccountButton.Click();
        }

        internal ISet<string> GetCustomerIds()
        {
            if (adminPanelLoginPage.Open().IsOnThisPage())
            {
                adminPanelLoginPage.EnterUsername("admin").EnterPassword("admin").SubmitLogin();
            }

            return customerListPage.Open().GetCustomerIds();
        }

    }
}

File: js-example/app/application.js
var webdriver = require('selenium-webdriver'),
    registration_page = require('../pages/registration_page'),
    admin_panel_login_page = require('../pages/admin_panel_login_page'),
    customer_list_page = require('../pages/customer_list_page');

class Application {

    constructor() {
        this.driver = new webdriver.Builder()
            .forBrowser("chrome")
            .build();
        this.registrationPage = new registration_page.RegistrationPage(this.driver);
        this.adminPanelLoginPage = new admin_panel_login_page.AdminPanelLoginPage(this.driver);
        this.customerListPage = new customer_list_page.CustomerListPage(this.driver);
    }

    quit() {
        this.driver.quit();
    }

    registerNewCustomer(customer) {
        this.registrationPage.open();
        this.registrationPage.firstnameInput().sendKeys(customer.firstname);
        this.registrationPage.lastnameInput().sendKeys(customer.lastname);
        this.registrationPage.address1Input().sendKeys(customer.address);
        this.registrationPage.postcodeInput().sendKeys(customer.postcode);
        this.registrationPage.cityInput().sendKeys(customer.city);
        this.registrationPage.selectCountry(customer.country);
        this.registrationPage.selectZone(customer.zone);
        this.registrationPage.emailInput().sendKeys(customer.email);
        this.registrationPage.phoneInput().sendKeys(customer.phone);
        this.registrationPage.passwordInput().sendKeys(customer.password);
        this.registrationPage.confirmedPasswordInput().sendKeys(customer.password);
        this.registrationPage.createAccountButton().click();
    }

    getCustomerIds() {
        var ap = this.adminPanelLoginPage;
        ap.open().onThisPageDo(function() {
            ap.enterUsername("admin").enterPassword("admin").submitLogin();
        });
        return this.customerListPage.open().getCustomerIds();
    }
}

exports.Application = Application;
